src/status_im/contexts/keycard/change_pin/events.cljs

```diff
@@ -0,0 +1,75 @@
+(ns status-im.contexts.keycard.change-pin.events
+  (:require [utils.i18n :as i18n]
+            [utils.re-frame :as rf]))
+
+(rf/reg-event-fx :keycard/change-pin.enter-current-pin
+ (fn [_]
+   {:fx [[:dispatch [:navigate-back]]
+         [:dispatch
+          [:open-modal :screen/keycard.pin.enter
+           {:title       (i18n/label :t/enter-current-keycard-pin)
+            :on-complete (fn [current-pin]
+                           (rf/dispatch [:navigate-back])
+                           (rf/dispatch [:keycard/change-pin.save-current-pin current-pin])
+                           (rf/dispatch [:keycard/change-pin.enter-new-pin]))}]]]}))
+
+(rf/reg-event-fx :keycard/change-pin.save-current-pin
+ (fn [{:keys [db]} [pin]]
+   {:db (assoc-in db [:keycard :change-pin :current-pin] pin)}))
```

Comment:
Since this is a secret, it should be stored as an instance of `MaskedData`. Even though we clean up if everything works correctly, it's good practice nonetheless. Masking secrets is also helpful to prevent accidental logging in the future.